docs/docs/03-data-engineering-roadmap.md

```diff
@@ -23,3 +23,80 @@ title: Data Engineering Roadmap
 ## Overview
 
 [!["Roadmap"](https://github.com/datastacktv/data-engineer-roadmap/raw/master/img/roadmap.png)](https://github.com/datastacktv/data-engineer-roadmap)
+
+### Software Engineering
```

Comment:
Might want to specify that this is for software engineer (data) or data platform engineer track, since some data engineers only use sql or gui tools, but they are still required to be knowledgeable about data modeling

https://www.thedataengineeringbook.online/docs/data-engineer#data-engineer-archtypes

Reply:
Thank you for useful review. I Will apply with my contribution but i confuse about my pull request that is Software engineer(data) or Data platform engineer.

Reply:
platform engineering is a cross between software engineer and devops 😉 

specific to the context of data engineering, software engineer (data) might be someone who create solutions that scale and run efficiently

but data platform engineer would be the one creating toolings with abstractions so other teams can easily consume/create stuff, and they are the one in charge of infrastructure

Reply:
สรุปคือผมควรจะแบ่งออกเป็นสกิลของทั้งสองสายของ Data Engineer ให้ชัดเจนเลยถูกมั้ยครับ เพราะหัวข้อที่ผมเข้าไปเขียนมันอยู่ในส่วนของความรู้พื้นฐานที่จำเป็นของ Data Engineer ผมเลยเข้าใจว่า Software Engineering ก็เป็นสิ่งที่ไม่ว่าจะเป็น software engineer (data) หรือ data platform engineer น่าจะต้องเอามาปรับใช้อยู่ดี

Reply:
nitpick นะอันนี้ ignore ได้

ทรง data engineer เอาแบบกว้างๆ มันแบ่งได้เป็น:
- สายปั่นดาต้า ใช้ sql อย่างเดียวก็รอด ---> analytics engineer / bi engineer 
- สายปั่นดาต้าขนาดใหญ่ๆ ต้อง optimize หนักๆ ---> software engineer (data)
- สายเซ็ตอินฟรา deploy โน่นนี่ให้ทีมใช้ ---> platform engineer

ประเภทบนสุด ไม่จำเป็นต้องรู้ software engineering

เอาจริงๆ สกิลมันก็คาบเกี่ยวกันอยู่ เกี่ยวมากเกี่ยวน้อยนี่อีกเรื่อง แต่มันก็มี data engineer ที่ปั่นดาต้าโดยใช้ sql ที่เขาก็ไม่จำเป็นต้องรู้ python เลย

Reply:
ผมแก้ไขแล้วครับ